src/shims/unix/linux/foreign_items.rs

```diff
@@ -72,6 +72,60 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             }
 
             // Threading
+            "prctl" => {
+                // We do not use `check_shim` here because `syscall` is variadic. The argument
+                // count is checked bellow.
+                this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+                if args.is_empty() {
+                    throw_ub_format!(
+                        "incorrect number of arguments for prctl: got 0, expected at least 1"
+                    );
+                }
+
+                // Since only two prctl calls are currently handled, and both expext just two arguments,
+                // it makes sense to handle validation here, outside of the match block below.
```

Comment:
No please don't do this -- you are now showing an error that says PR_SET_NAME befre even checking whether that's the op we are executing.

Reply:
My mistake, forgot that I included the name into the message.